### DISCUSSION TITLE
Crunch compression support

### DISCUSSION THREAD

**Discussion by Artein** [Elringus/SpriteDicing#37] [General]
Hi, Artyom!
Hope you are doing well.

How it is possible to enable crunch compression for texture/atlas ?
Right now I see in Memory Profile that texture is RGBA32 and allocates 10MB (that is too much)

Ps. also MemoryProfiler with snapshot from Android divice shows the same 10MB. Is it okay, if I see such compression ratio in Editor?
<img width="406" alt="SCR-20231027-tsma" src="https://github.com/Elringus/SpriteDicing/assets/1109722/9dd83ae1-7bed-4da3-bad6-4c8e1f65217d">

**Comment by elringus** [comment]
Hey, Just use import settings on the generated atlas texture, same as with any other textures. The settings will be preserved when re-generating the atlas (make sure to use the latest version of the package, as this was added in one of the recent updates).

**Reply by Artein** [reply]
Thank you, for the prompt answer!

Yes, I did that, as well as tuning default settings in the package (thought maybe there some texture regeneration during builds). Plus, I changed sprite format to "Sprite 2D + UI" instead of Default.
But even if I enable crunch compression in import settings, I see no changes in the final texture in build.

FYI, I am using Addressables. May they influence somehow?

Here what I have in Editor:
<img width="375" alt="SCR-20231028-kmot" src="https://github.com/Elringus/SpriteDicing/assets/1109722/0c806334-502a-4337-b6ab-d0d83df37aef"><img width="375" alt="SCR-20231028-kmra" src="https://github.com/Elringus/SpriteDicing/assets/1109722/6ef7fb44-f26b-4c0b-8111-c7d25c9e738f">

**Reply by elringus** [reply]
Maybe the platform doesn't support the compression? Not sure. Does it work on other platforms?

**Reply by elringus** [reply]
Also, texture type should be left default. Sprite mode is intended to be used as a source for sprite renderers and UI images.

**Reply by Artein** [reply]
Yes, I have other crunched textures. The platform is Android

**Reply by elringus** [reply]
Double-checked and it seem to work fine on my end. In the example project, try building it as-is, the build size will be 250MB. Then set up max. compression on the 3 animation atlases and rebuild. The build size is now 60MB.

![c83b15af067082040a5cefeffa8dcd41](https://github.com/Elringus/SpriteDicing/assets/2056864/d795c3a9-ff58-4f9e-8b47-23fed392349b)
.

**Reply by Artein** [reply]
Thank you for helping me, Artyom!

For now, I am curious why `BuildReport` shows that, meanwhile having the texture in the build (it is regenerated on the fly?).
And as you can see, the size seems to be correct. How interesting... 

Should I also mark the texture as addressable and add to some `Naninovel-XXX` group?

<img width="700" alt="SCR-20231028-mdtk" src="https://github.com/Elringus/SpriteDicing/assets/1109722/5c17dbb1-0100-4783-86aa-5544df5df231">

**Reply by elringus** [reply]
I don't have much experience with addressables, but iirc they should track the references automatically. So, if you already have the diced sprites "addressed", the system should include associated atlas automatically. Just make sure to not include (or otherwise implicitly reference) original sprite textures.

**Reply by Artein** [reply]
Yes, I don't reference the texture manually (only DicedSpriteAtlas). 

It seems like the issue is that the texture is not packed within the build (even though I manually enable it in addressables). The first guess is that it is recreating somehow at the runtime. Will continue my investigation...
